internal/provider/random_ip.go

```diff
@@ -0,0 +1,194 @@
+)
+
+var (
+	_ resource.Resource = (*ipResource)(nil)
+)
+
+func NewIPResource() resource.Resource {
+	return &ipResource{}
+}
+
+type ipResource struct{}
+
+func (r *ipResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_ip"
+}
+
+func (r *ipResource) Schema(ctx context.Context, req resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "The resource `random_ip` generates a random IP address from a given CIDR range based on the " +
+			"address type specified.",
+		Attributes: map[string]schema.Attribute{
+			"keepers": schema.MapAttribute{
+				Description: "Arbitrary map of values that, when changed, will trigger recreation of " +
+					"resource. See [the main provider documentation](../index.html) for more information.",
+				ElementType: types.StringType,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Map{
+					mapplanmodifiers.RequiresReplaceIfValuesNotNull(),
+				},
+			},
+			"address_type": schema.StringAttribute{
+				Description: "A string indicating the type of IP address to generate. Valid values are `ipv4` and `ipv6`.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"ipv4", "ipv6"}...),
+				},
+			},
+			"cidr_range": schema.StringAttribute{
+				Description: "A CIDR range from which to allocate the IP address.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"result": schema.StringAttribute{
+				Description: "The random IP address allocated from the CIDR range.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"id": schema.StringAttribute{
+				Description: "A static value used internally by Terraform, this should not be referenced in configurations.",
+				Computed:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+		},
+	}
+}
+
+func (r *ipResource) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
+	var plan ipModelV0
+
+	diags := req.Plan.Get(ctx, &plan)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	addressType := plan.AddressType.ValueString()
+	cidrRange := plan.CIDRRange.ValueString()
+
+	// Generate a random IP address from a given CIDR range.
+	ip, err := getRandomIP(cidrRange, addressType)
+	if err != nil {
+		resp.Diagnostics.AddError(
+			"Create Random IP error",
+			"There was an error when generating the random IP address.\n\n"+
+				diagnostics.RetryMsg+
+				"Original Error: "+err.Error(),
+		)
+		return
+	}
+
+	u := &ipModelV0{
+		ID:          types.StringValue("-"),
```

Comment:
What internal ID is preferred? There is no consistent usage across the provider.